pkg/states/state_manager.go

```diff
@@ -0,0 +1,35 @@
+package states
+
+import "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+
+func Deactivating(userSignup *v1alpha1.UserSignup) bool {
+	return contains(userSignup.Spec.States, v1alpha1.UserSignupStateDeactivating)
+}
+
+func SetUserSignupState(userSignup *v1alpha1.UserSignup, state v1alpha1.UserSignupState) {
```

Comment:
Maybe AddUserSignupSrate() instead? Just to avoid confusion since this function do not replace the current states and only appends a new one.

Or maybe we should avoid generic exported functions here at all because some states should replace specific states while others can be just added or removed.

So maybe something like SetDeactivatingState(), UnsetDeactivatingState(), SetDeactivated(), SetApproved(), etc.

WDYT?

Reply:
I like the idea of specific functions like `SetDeactivating()`, `SetVerificationRequired()` etc, but perhaps these functions could take a `bool` param to indicate whether the state should be set or unset?

Reply:
Yep. +1 to using a bool param.